### PR TITLE
[resoto][feat] CDK construct

### PR DESCRIPTION
**Pull request by meln1k**
# Description

Added a CDK construct for a one command resoto deployment to AWS.

# To-Dos

- [ ] Store container data on a different volume 

- [ ] Document new or updated functionality (someengineering/resoto.com#XXXX)

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: cdk/lib/resoto-stack.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    super(scope, id, props);
+
+    const vpc = new ec2.Vpc(this, 'resoto-vpc', {
+      maxAzs: 1,
+      subnetConfiguration: [{
+          name: 'public', 
+          cidrMask: 24,
+          subnetType: ec2.SubnetType.PUBLIC,
+        }]
+    });
```
Comment (aquamatthias) [inline cdk/lib/resoto-stack.ts:17]: I am wondering if we should support deploying it into an existing vpc?

File: cdk/lib/resoto-stack.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      securityGroup: resotocoreSG,
+      instanceType: ec2.InstanceType.of(
+        ec2.InstanceClass.T3,
+        ec2.InstanceSize.LARGE
+      ),
+      machineImage: image,
+      blockDevices: [
+        {
+          deviceName: '/dev/xvda', // make the root volume bigger
+          volume: ec2.BlockDeviceVolume.ebs(50),
```
Comment (aquamatthias) [inline cdk/lib/resoto-stack.ts:78]: I guess the size should be configurable?

File: cdk/lib/resoto-stack.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        },
+      ],
+      keyName: 'nm-resoto-ec2-keypair',
```
Comment (aquamatthias) [inline cdk/lib/resoto-stack.ts:81]: Where is this keypair coming from?

File: cdk/lib/resoto-stack.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      'sudo mv docker-compose-$(uname -s)-$(uname -m) /usr/local/bin/docker-compose',
+      'sudo chmod -v +x /usr/local/bin/docker-compose',
+      'sudo systemctl enable docker.service',
+      'sudo systemctl start docker.service',
+      // resoto setup
+      'mkdir -p resoto/dockerV2',
+      'cd resoto',
+      'curl -o docker-compose.yaml https://raw.githubusercontent.com/someengineering/resoto/2.4.3/docker-compose.yaml',
+      'curl -o dockerV2/prometheus.yml https://raw.githubusercontent.com/someengineering/resoto/2.4.3/dockerV2/prometheus.yml',
+      'docker-compose up -d',
```
Comment (aquamatthias) [inline cdk/lib/resoto-stack.ts:59]: This means it is docker version 1 - is it possible to user docker version 2?

**Review by aquamatthias — COMMENTED** [review]
Great start - thanks.
I think this content should be placed into a separate repository.

**Comment by lloesche** [comment]
My two cents: I'm very much against all of this single point of failure deployment model. Would MUCH more prefer deployment on EKS where failures are automatically resolved and data is persisted onto dedicated EBS volumes. If somebody can't use K8S they're likely the wrong user for Resoto anyways.

File: cdk/lib/resoto-stack.ts (hunk 4 above)
Comment (meln1k) [reply cdk/lib/resoto-stack.ts]: that's not straightforward using the amazon image.

File: cdk/lib/resoto-stack.ts (hunk 3 above)
Comment (meln1k) [reply cdk/lib/resoto-stack.ts:69]: needs to be configurable

File: cdk/lib/resoto-stack.ts (hunk 2 above)
Comment (meln1k) [reply cdk/lib/resoto-stack.ts:62]: yep

**Comment by meln1k** [comment]
@lloesche @scapecast if k8s is what you wish, then so be it :) I will update the CDK construct to create a new EKS cluster in case our users do not have one yet, and then deploy a helm chart on it.

**Comment by aquamatthias** [comment]
Moved to a separate repository: https://github.com/someengineering/resoto-cdk

Closing this PR.

**Pull request closed without merging**